subsys/nrf_security/src/drivers/cracen/cracenpsa/src/blkcipher.c

```diff
@@ -0,0 +1,693 @@
+				sx_status = sx_blkcipher_resume_state(&operation->cipher);
+				if (sx_status) {
+					return silex_statuscodes_to_psa(sx_status);
+				}
+			} else {
+				psa_status_t r = initialize_cipher(operation);
+
+				if (r != PSA_SUCCESS) {
+					return r;
+				}
+			}
+
+			if (operation->unprocessed_input_bytes) {
+				__ASSERT_NO_MSG(operation->unprocessed_input_bytes ==
+						operation->blk_size);
+				sx_status = sx_blkcipher_crypt(
+					&operation->cipher, operation->unprocessed_input,
+					operation->unprocessed_input_bytes, output);
+				if (sx_status) {
+					return silex_statuscodes_to_psa(sx_status);
+				}
+
+				operation->unprocessed_input_bytes = 0;
+				/* Adjust output pointer. */
+				output += (uint32_t)operation->blk_size;
+			}
+
+			if (block_bytes) {
+				sx_status = sx_blkcipher_crypt(&operation->cipher, input,
+							       block_bytes, output);
+				if (sx_status) {
+					return silex_statuscodes_to_psa(sx_status);
+				}
+			}
+
+			sx_status = sx_blkcipher_save_state(&operation->cipher);
+			if (sx_status) {
+				return silex_statuscodes_to_psa(sx_status);
+			}
+
+			sx_status = sx_blkcipher_wait(&operation->cipher);
+			if (sx_status) {
+				return silex_statuscodes_to_psa(sx_status);
+			}
+
+			*output_length += total_output;
+		}
+	}
+
+	/* Store unprocessed bytes until next update or finalization of crypto
+	 * operation.
+	 */
+	if (block_bytes < input_length) {
+		__ASSERT_NO_MSG(operation->unprocessed_input_bytes == 0);
+		memcpy(operation->unprocessed_input, input + block_bytes,
+		       input_length - block_bytes);
+		operation->unprocessed_input_bytes = input_length - block_bytes;
+	}
+
+	(void)status;
+	return PSA_SUCCESS;
+}
+
+psa_status_t cracen_cipher_finish(cracen_cipher_operation_t *operation, uint8_t *output,
+				  size_t output_size, size_t *output_length)
+{
+	__ASSERT_NO_MSG(output != NULL);
+	__ASSERT_NO_MSG(output_length != NULL);
+
+	int sx_status;
+
+	*output_length = 0;
+
+	if (operation->unprocessed_input_bytes == 0 && operation->alg != PSA_ALG_CBC_PKCS7) {
+		return PSA_SUCCESS;
+	}
+
+	sx_status = SX_ERR_UNINITIALIZED_OBJ;
+
+	/* sxsymcrypt doesn't support context saving for ECB, as each encrypted block is
+	 * independent from the previous one we just encrypt the so far available full
+	 * blocks.
+	 */
+	if (IS_ENABLED(PSA_NEED_CRACEN_ECB_NO_PADDING_AES)) {
+		if (operation->alg == PSA_ALG_ECB_NO_PADDING) {
+			return cracen_cipher_crypt_ecb(&operation->keyref,
+						       operation->unprocessed_input,
+						       operation->unprocessed_input_bytes, output,
+						       output_size, output_length, operation->dir);
+		}
+	}
+
+	if (operation->initialized) {
+		sx_status = sx_blkcipher_resume_state(&operation->cipher);
+		if (sx_status) {
+			return silex_statuscodes_to_psa(sx_status);
+		}
+	} else {
+		psa_status_t r = initialize_cipher(operation);
+
+		if (r != PSA_SUCCESS) {
+			return r;
+		}
+	}
+
+	if (IS_ENABLED(PSA_NEED_CRACEN_CBC_PKCS7_AES)) {
```

Comment:
Follow up task: Convert to ifdef

Reply:
We have intentionally changed from ifdef to IS_ENABLED.

I can't remember the details, but it was something about a warning.

In any case, using IS_ENABLED is good practice in this context.